app/controllers/KairosDbProxyController.java

```diff
@@ -173,6 +187,29 @@ public KairosDbProxyController(
         }
     }
 
+    MetricsQuery addMergeAggregator(final MetricsQuery metricsQuery) {
+        final List<Metric> newMetrics = new ArrayList<>();
+        for (Metric metric : metricsQuery.getMetrics()) {
+            List<Aggregator> newAggregators = new ArrayList<>();
+            if (!metric.getAggregators().isEmpty() && !(metric.getAggregators().get(0).getName()).equals("merge")
+                    && metric.getAggregators().stream().noneMatch(
+                            aggregator -> NONHISTAGGREGATORS.contains(aggregator.getName()))) {
+                final Optional<Sampling> originalSampling = metric.getAggregators().get(0).getSampling();
+                newAggregators.add(ThreadLocalBuilder.build(Aggregator.Builder.class,
+                        b -> b.setName("merge").setSampling(originalSampling.map(
+                                sampling -> ThreadLocalBuilder.clone(sampling, Sampling.Builder.class)).orElse(null))));
+                newAggregators.addAll(metric.getAggregators());
+            } else {
+                newAggregators = metric.getAggregators();
+            }
+            final ImmutableList<Aggregator> finalNewAggregators = ImmutableList.copyOf(newAggregators);
+            newMetrics.add(ThreadLocalBuilder.clone(metric, Metric.Builder.class, b->b.setAggregators(finalNewAggregators)));
```

Comment:
We could avoid copying the `Metrics` object if the `newAggregators` was not modified. For example, marking `newAggregators` as `@Nullable` and initially setting it to `null` and not setting it in the `else` branch at all. Now we know whether we need to modify the metric or not.